CDCgov/cdc-open-viz
# [DEV-9416] X-axis improvements for time series charts

## [DEV-9416]

This does four main things: 

* Makes abbreviated months with period work correctly for May
* Adds option to show each year only once in axis labels
* Padding only applies to left side
* Calculates the width of the chart correctly so gridlines extend to the end of the x-axis

## Testing Steps

Use the config attached to the ticket.

* Note that May is correctly formatted
* Toggle the "show years once" button in the date/category axis and see it has the desired effect
* Adjust the padding option in the date/category axis and see that it applies only to the left side
* Note that gridlines extend to the end of the x-axis

## Self Review

- I have added testing steps for reviewers
- I have commented my code, particularly in hard-to-understand areas
- My changes generate no new warnings
- New and existing unit tests are passing

## Screenshots (if applicable)

![Screenshot 2024-10-09 at 9 59 25 AM](https://github.com/user-attachments/assets/20615384-28f8-419c-bb15-3e6d50565d1f)
